File: field-force-cpp/field-force-cpp/board/fire.cpp
#include "fire.h"

Fire::Fire() { }

Fire::Fire(char _id, int _x_pos, int _y_pos)
{
	id = _id;
	x_pos = _x_pos;
	y_pos = _y_pos;
}

Fire::Fire(char _id, int _x_pos, int _y_pos, int _duration)
{
	id = _id;
	x_pos = _x_pos;
	y_pos = _y_pos;
	duration = _duration;
}

File: field-force-java/src/Game/Fire.java
package Game;

public class Fire extends MapObject {
    private int duration = 2;

    public Fire(char id, int x, int y){
        super(id, x, y);
    }

    public int reduceDuration(){
        return duration--;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }
}
